### PR TITLE
hal: gd32f470xx: Add pinconfigs file

### PR DESCRIPTION
**Pull request by duchesne9al**
Add pinconfigs file for gd32f470xx with associated generated pinctrl headers

Related PR  : https://github.com/zephyrproject-rtos/zephyr/pull/46894.

Signed-off-by: Alexandre Duchesne <alexandre.duchesne@rtone.fr>

### CODE REVIEW

**Comment by nandojve** [comment]
Hi @duchesne9al,

Thank you for your PR. I noted that Gigadevice have a newer library, version 3.0.0, which supports F470.
I think it is important update current lib to newer version to make sure code have full support to F470.

**Review by gmarull — CHANGES_REQUESTED** [review]
please to ease review, add 2 commits: one that adds the yaml pinconfig file and another one for the autogenerated files. Also please open a PR in Zephyr updating HAL sha (and adding support for f470?)

File: pinconfigs/gd32f470xx.yml [hunk 1]
```diff
@@ -0,0 +1,1229 @@
...
+# - 100 pins: V
+#
+# Memory codes:
+#
+# - 512Kb  Flash, 256Kb SRAM: E
+# - 1024Kb Flash, 256Kb SRAM: G
+# - 2048Kb Flash, 512Kb SRAM: I
+# - 3072Kb Flash, 256Kb SRAM: K
+#
+# Copyright (c) 2022, Rtone.
```
Comment (gmarull) [inline pinconfigs/gd32f470xx.yml:19]: this is mostly a copy of the f450xx file, so copyrights should be kept.

Comment (duchesne9al) [reply pinconfigs/gd32f470xx.yml]: > this is mostly a copy of the f450xx file, so copyrights should be kept.

Sure, no problem.

File: pinconfigs/gd32f470xx.yml [hunk 2]
```diff
@@ -0,0 +1,1228 @@
...
+# - 100 pins: V
+#
+# Memory codes:
+#
+# - 512Kb  Flash, 256Kb SRAM: E
+# - 1024Kb Flash, 256Kb SRAM: G
+# - 2048Kb Flash, 512Kb SRAM: I
+# - 3072Kb Flash, 256Kb SRAM: K
+#
+# SPDX-License-Identifier: Apache 2.0
```
Comment (gmarull) [inline pinconfigs/gd32f470xx.yml:19]: there's no copyright now

Comment (duchesne9al) [reply pinconfigs/gd32f470xx.yml]: oups sry

**Pull request merged**